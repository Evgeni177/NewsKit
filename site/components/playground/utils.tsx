import React from 'react';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const isComponent = (value: any): value is React.ComponentType =>
  Boolean(value && (value.$$typeof || value.displayName));

export const withProps = (
  Component: React.ComponentType,
  props: Record<string, unknown>,
  displayName?: string,
) => {
  const Comp: React.FC = p => <Component {...props} {...p} />;
  Comp.displayName = displayName || Component.displayName;
  return Comp;
};
