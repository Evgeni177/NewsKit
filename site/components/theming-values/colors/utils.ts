import {Theme, ThemeProp} from 'newskit';
import {ThemeColor} from './types';
import {getContrast, getBrightness, getContrastRating} from './color-utils';

export const getBorderRadius = ({
  first,
  last,
  theme,
}: {
  first?: boolean;
  last?: boolean;
  theme?: Theme;
}) => {
  const fn = ({
    theme: {
      borderRadius: {borderRadiusRounded020: radius},
    },
  }: ThemeProp) =>
    [
      first ? radius : 0,
      first ? radius : 0,
      last ? radius : 0,
      last ? radius : 0,
    ].join(' ');

  if (theme) {
    return fn({theme});
  }

  return fn;
};

export const getTextColor = ({
  theme,
  value,
}: // brightness,
{theme: Theme} & Pick<ThemeColor, 'brightness' | 'value'>) =>
  // brightness < 0.4 ? theme.colors.inkInverse : theme.colors.inkContrast;
  getContrast(value, theme.colors.inkInverse) <
  getContrast(value, theme.colors.inkContrast)
    ? theme.colors.inkInverse
    : theme.colors.inkContrast;

export const getColorGroup = (name: string) => (/[a-z]+/.exec(name) || [])[0];

export const mapColorObjects = (
  theme: Theme,
  paletteColorMap?: Record<string, ThemeColor[]>,
) => ([name, value]: [string, string]) => {
  const brightness = getBrightness(value);
  const contrastRatio = getContrast(value, theme.colors.interface010);
  const contrastRating = getContrastRating(contrastRatio);

  return {
    name,
    value,
    group: getColorGroup(name),
    brightness,
    contrastRating,
    contrastRatio,
    parentColor:
      paletteColorMap && paletteColorMap[value]
        ? paletteColorMap[value][0]
        : undefined,
  } as ThemeColor;
};

export const groupColorObjects = (
  acc: Record<string, ThemeColor[]>,
  color: ThemeColor,
) => {
  // eslint-disable-next-line no-multi-assign
  const group = (acc[color.group] = acc[color.group] || []);
  group.push(color);
  return acc;
};

export const createPaletteColorMap = (paletteColors: ThemeColor[]) =>
  paletteColors.reduce(
    (acc, color) => {
      // eslint-disable-next-line no-multi-assign
      const accGroup = (acc[color.value] = acc[color.value] || []);
      accGroup.push(color);
      return acc;
    },
    {} as Record<string, ThemeColor[]>,
  );
