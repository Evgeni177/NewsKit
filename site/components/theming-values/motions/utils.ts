export const CANVAS_SIZE = 56;

const parseCubicBezier = (value: string) =>
  value
    .replace('cubic-bezier(', '')
    .replace(')', '')
    .split(',')
    .map(v => Number.parseFloat(v.trim()));

// NEXT 2 FUNCTIONS ARE COPIED FROM http://dbaron.org/css/timing-function-graphs
export function bezier(x1: number, y1: number, x2: number, y2: number) {
  // Cubic bezier with control points (0, 0), (x1, y1), (x2, y2), and (1, 1).
  function xForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * x1 + 3 * omt * t * t * x2 + t * t * t;
  }
  function yForT(t: number) {
    const omt = 1 - t;
    return 3 * omt * omt * t * y1 + 3 * omt * t * t * y2 + t * t * t;
  }
  function tForX(x: number) {
    // Binary subdivision.
    let mint = 0;
    let maxt = 1;
    for (let i = 0; i < 30; ++i) {
      const guesst = (mint + maxt) / 2;
      const guessx = xForT(guesst);
      if (x < guessx) maxt = guesst;
      else mint = guesst;
    }
    return (mint + maxt) / 2;
  }
  return function bezierClosure(x: number) {
    if (x === 0) return 0;
    if (x === 1) return 1;
    return yForT(tForX(x));
  };
}

export const updateCanvas = (
  ctx: CanvasRenderingContext2D,
  timingFn: string,
  color: string,
) => {
  let x1 = 0;
  let y1 = 0;
  let x2 = 0;
  let y2 = 0;
  switch (timingFn) {
    case 'ease':
      x1 = 0.25;
      y1 = 0.1;
      x2 = 0.25;
      y2 = 1.0;
      break;
    case 'linear':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-in':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 1.0;
      y2 = 1.0;
      break;
    case 'ease-out':
      x1 = 0.0;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    case 'ease-in-out':
      x1 = 0.42;
      y1 = 0.0;
      x2 = 0.58;
      y2 = 1.0;
      break;
    default:
      [x1, y1, x2, y2] = parseCubicBezier(timingFn);
      break;
  }

  const tf = bezier(x1, y1, x2, y2);

  const gCurrentTFPoints = [];

  for (let x = 0; x <= CANVAS_SIZE; ++x) {
    gCurrentTFPoints[x] = tf(x / CANVAS_SIZE) * CANVAS_SIZE;
  }

  ctx.clearRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);

  // Draw the graph
  ctx.strokeStyle = color;
  ctx.lineWidth = 2;
  ctx.beginPath();
  ctx.moveTo(0, CANVAS_SIZE - gCurrentTFPoints[0]);
  for (let x = 1; x <= CANVAS_SIZE; ++x) {
    ctx.lineTo(x, CANVAS_SIZE - gCurrentTFPoints[x]);
  }
  ctx.stroke();
};

export const percentFrom = (total: number, value: number) =>
  Math.floor(value / (total / 100));

export const getTotalDuration = (stayDuration: number, maxDuration: number) =>
  2 * stayDuration + 2 * maxDuration;
