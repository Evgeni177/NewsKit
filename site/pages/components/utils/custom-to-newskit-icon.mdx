import {LegacyBlock} from '../../../components/legacy-block';
import Layout from '../../../components/layout';
import {Code, CodeFromFile} from '../../../components/code';
import Prop from '../../../components/prop';

export default Layout;

# customToNewsKitIcon()

## Syntax

<Code>{'customToNewsKitIcon(displayName, CustomIcon, overrides)'}</Code>

## Overview

Transform SVG markup into a Newskit icon that works well with theme tokens and `getFromTheme` util functions.

## Parameters

<LegacyBlock marginTop="sizing040">
  <Prop name="displayName" type="string" required>
    The display name of the icon. It is required an icon name to start either
    with <b>`IconFilled`</b> or <b>`IconOutlined`</b> prefix, depending on the
    imported icon type. <br />
    E.g: If importing an <b>MenuBook</b> icon, the final icon display name should
    be <b>`IconFilledMenuBook`</b> or <b>`IconOutlinedMenuBook`</b>
  </Prop>
  <Prop name="CustomIcon" type="React.ComponentType<SvgProps>" required>
    NewsKit Svg component with SVG passed to it's children, props should be
    spread on the root Svg component, see example below.
  </Prop>
  <Prop name="overrides" type='SvgProps["overrides"]'>
    Default <a href="/components/icons#props">override props</a> passed to
    the transformed icon. Overrides can still be passed to the icon to overwrite
    these values at render time.
  </Prop>
</LegacyBlock>

## Return value

Returns a NewsKit icon.

## Example

<CodeFromFile path="examples/utils/custom-to-newskit-icon.tsx" />
