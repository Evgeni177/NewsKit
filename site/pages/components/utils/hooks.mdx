import Layout from '../../../components/layout';
import {Code} from '../../../components/code';
import {LegacyBlock} from '../../../components/legacy-block';
import Prop from '../../../components/prop';
import {Link} from 'newskit';

export default Layout;

# Hooks

## useMediaQueryObject

### Overview

`useMediaQueryObject` hook handles scenarios in which you want to render component based on media query breakpoints.
This hooks also responds to the window resizing and returns the appropriate value for the new window size.

`useMediaQueryObject` returns the value for the current breakpoint from the provided media query object ( `MQ<T>` )

Note: The hook needs `<MediaQueryProvider>` to be added to an ancestor of the component in which it's used.

### Example

A common use case for using `useMediaQueryObject` is when you want to change component property based on media query.
In the example below, the `<Card />` component changes its layout depending on screen the size.

<Code>
  {`const mqLayout = {
    xs: 'vertical',
    sm: 'vertical',
    md: 'horizontal',
    lg: 'horizontal',
    xl: 'horizontal',
}
const layout = useMediaQueryObject(mqLayout);
<Card layout={layout} /> `}
</Code>

Note: In theory `useMediaQueryObject` could be used with any `MQ<T>`, however, we recommend using it only for props that don't accept `MQ<T>` as a value. Most CSS-based props and overrides already support `MQ<T>` objects.
When you create your own components and want to apply responsive behaviour you should try to use <Link href="/utils/get-from-theme">getters</Link> and <Link href="/utils/get-defaults">defaults</Link> functions.

DO NOT DO THIS:

<Code>
  {`const stylePresets = {
    xs: 'dividerPrimary',
    md: 'horizontal',
}
const dividerStylePreset = useMediaQueryObject(stylePresets);
<Divider overrides={{stylePreset: dividerStylePreset}} />`}
</Code>

## useBreakpointKey

### Overview

`useBreakpointKey` has a similar utility as `useMediaQueryObject`, it's intended usage is where you want to know the currently active breakpoint key `xs | sm | md | lg | xl`.

### Example

<Code>
  {`const breakpointKey = useBreakpointKey();
// return XL when screen has width XL ( 1440px and above )`}
</Code>

`<Card />` component use case

<Code>
  {`const breakpointKey = useBreakpointKey();
  const layout == breakpointKey === 'xs || breakpointKey === 'sm' ? 'vertical' : 'horizontal';
<Card layout={layout} />
// the <Card will render verticaly on XS and SM screens and Horizontaly on the rest of breakpoints`}

</Code>
