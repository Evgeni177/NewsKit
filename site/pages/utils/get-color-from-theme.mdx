import {LegacyBlock} from '../../components/legacy-block';
import Layout from '../../components/layout';
import {Code, CodeFromFile} from '../../components/code';
import Prop from '../../components/prop';

export default Layout;

# getColorFromTheme()

## Syntax

<Code>{'getColorFromTheme(defaultToken, customProp)(props)'}</Code>

## Overview

Retrieve a color token from the current theme. Color tokens are [CSS color values](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value) to be used for CSS styling.

## Parameters

<LegacyBlock marginTop='sizing040'>
  <Prop name='defaultToken' type='ThemeToken'>
    Is the name of the color token to be retrieved. If provided, it is used as an object key on the color section of the theme to get the token value.
  </Prop>
  <Prop name='customProp' type="Exclude<keyof Props, 'theme'>">
    Is the name of the component prop in which a color token name can be provided by the end-user of the component. If provided, the color to be retrieved will come from the component prop.
  </Prop>
  <Prop name='props' type='Props'>
    Is the props of the component including the theme. If the function is used inside a styled component template literal, or a css block, Emotion.js will call the function first with the component props passed in.
  </Prop>
</LegacyBlock>

## Return value

Returns a color token if valid attributes are received.

Returns an empty string if invalid attributes or no attributes are provided.

## Example

<CodeFromFile path="examples/utils/get-color-from-theme.tsx"/>

In this example the `Paragraph` component will include the color value from `highlight` found in the theme if the `Paragraph` is used with no props:

<Code>{'<Paragraph />'}</Code>

However, it will include value from `focus` if the `Paragraph` is used with the `color` prop and the value passed in is `focus`:

<Code>{'<Paragraph color="focus" />'}</Code>
