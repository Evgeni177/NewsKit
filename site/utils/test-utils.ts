import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
