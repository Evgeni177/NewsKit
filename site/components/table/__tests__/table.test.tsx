import React from 'react';
import {renderToFragmentWithTheme} from '../../../utils/test-utils';
import {Table} from '..';

describe('Table', () => {
  test('renders with default styles', () => {
    const props = {
      columns: ['Col', 'Column'],
      rows: [
        {
          col: 'col',
          column: 'column',
        },
        {
          col: 'col',
          column: 'column',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });
  test('wraps content with styledFlag if the column is either Name, Type or Default styles', () => {
    const props = {
      columns: ['Name', 'Type', 'Default'],
      rows: [
        {
          name: 'name',
          type: 'type',
          default: 'default',
        },
        {
          name: 'name',
          type: 'type',
          default: 'default',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });

  test('wraps content with styledFlag if the column is either Name, Type or Default styles', () => {
    const props = {
      columns: ['Name', 'Type', 'Default'],
      rows: [
        {
          name: 'name',
          type: 'type',
          default: 'default',
        },
        {
          name: 'name',
          type: 'type',
          default: 'default',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders IconFilledCheckCircle if we pass true for a `Required` columm', () => {
    const props = {
      columns: ['Column', 'Required'],
      rows: [
        {
          column: 'column',
          required: true,
        },
        {
          column: 'column',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders number values automatically if undefined', () => {
    const props = {
      columns: ['Item', 'Order', 'Description'],
      rows: [
        {
          description: 'one',
        },
        {
          description: 'two',
        },
        {
          description: 'three',
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders links correctly', () => {
    const props = {
      columns: ['Component'],
      rows: [
        {
          component: 'Button',
        },
        {
          component: <span>test</span>,
        },
      ],
    };
    const fragment = renderToFragmentWithTheme(Table, props);
    expect(fragment).toMatchSnapshot();
  });
});
