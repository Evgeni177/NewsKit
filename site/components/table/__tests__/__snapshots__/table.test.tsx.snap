// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders IconFilledCheckCircle if we pass true for a \`Required\` columm 1`] = `
<DocumentFragment>
  .emotion-13 {
  overflow-x: auto;
}

.emotion-12 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  margin-bottom: 8px;
  height: 40px;
  text-align: left;
  vertical-align: middle;
  min-width: 64px;
}

.emotion-6 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-2 {
  padding: 8px 8px 8px 0;
}

.emotion-3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  fill: #0a782a;
  vertical-align: unset;
  display: inline-block;
}

.emotion-3.emotion-3 {
  width: 24px;
  height: 24px;
}

<div
    class="emotion-13"
  >
    <table
      class="emotion-12"
    >
      <thead>
        <tr>
          <th
            class="emotion-0"
          >
            Column
          </th>
          <th
            class="emotion-0"
          >
            Required
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="emotion-6"
        >
          <td
            class="emotion-2"
          >
            column
          </td>
          <td
            class="emotion-2"
          >
            <svg
              aria-hidden="true"
              class="emotion-3 emotion-4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </td>
        </tr>
        <tr
          class="emotion-6"
        >
          <td
            class="emotion-2"
          >
            column
          </td>
          <td
            class="emotion-2"
          >
            <svg
              aria-hidden="true"
              class="emotion-3 emotion-4"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table renders with default styles 1`] = `
<DocumentFragment>
  .emotion-9 {
  overflow-x: auto;
}

.emotion-8 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  margin-bottom: 8px;
  height: 40px;
  text-align: left;
  vertical-align: middle;
  min-width: 64px;
}

.emotion-4 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-2 {
  padding: 8px 8px 8px 0;
}

<div
    class="emotion-9"
  >
    <table
      class="emotion-8"
    >
      <thead>
        <tr>
          <th
            class="emotion-0"
          >
            Col
          </th>
          <th
            class="emotion-0"
          >
            Column
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="emotion-4"
        >
          <td
            class="emotion-2"
          >
            col
          </td>
          <td
            class="emotion-2"
          >
            column
          </td>
        </tr>
        <tr
          class="emotion-4"
        >
          <td
            class="emotion-2"
          >
            col
          </td>
          <td
            class="emotion-2"
          >
            column
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table wraps content with styledFlag if the column is either Name, Type or Default styles 1`] = `
<DocumentFragment>
  .emotion-42 {
  overflow-x: auto;
}

.emotion-41 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  margin-bottom: 8px;
  height: 40px;
  text-align: left;
  vertical-align: middle;
  min-width: 64px;
}

.emotion-21 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-8 {
  padding: 8px 8px 8px 0;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 32px;
  padding: 8px 12px;
  cursor: default;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
}

.emotion-6 svg {
  width: 16px;
  height: 16px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    max-width: 100%;
  }
}

.emotion-3 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-42"
  >
    <table
      class="emotion-41"
    >
      <thead>
        <tr>
          <th
            class="emotion-0"
          >
            Name
          </th>
          <th
            class="emotion-0"
          >
            Type
          </th>
          <th
            class="emotion-0"
          >
            Default
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="emotion-21"
        >
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      name
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      type
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      default
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="emotion-21"
        >
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      name
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      type
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      default
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Table wraps content with styledFlag if the column is either Name, Type or Default styles 2`] = `
<DocumentFragment>
  .emotion-42 {
  overflow-x: auto;
}

.emotion-41 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  margin-bottom: 8px;
  height: 40px;
  text-align: left;
  vertical-align: middle;
  min-width: 64px;
}

.emotion-21 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-8 {
  padding: 8px 8px 8px 0;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 32px;
  padding: 8px 12px;
  cursor: default;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
}

.emotion-6 svg {
  width: 16px;
  height: 16px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-4px/2);
  margin-right: calc(-4px/2);
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(4px/2);
  margin-right: calc(4px/2);
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    max-width: 100%;
  }
}

.emotion-3 {
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.emotion-3::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: calc(-0.3642857142857143em + 1.2px);
}

.emotion-3::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-42"
  >
    <table
      class="emotion-41"
    >
      <thead>
        <tr>
          <th
            class="emotion-0"
          >
            Name
          </th>
          <th
            class="emotion-0"
          >
            Type
          </th>
          <th
            class="emotion-0"
          >
            Default
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="emotion-21"
        >
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      name
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      type
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      default
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class="emotion-21"
        >
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      name
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      type
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            class="emotion-8"
          >
            <div
              class="emotion-7"
            >
              <div
                class="emotion-6"
                data-testid="flag"
              >
                <div
                  class="emotion-5"
                >
                  <div
                    class="emotion-4"
                  >
                    <span
                      class="emotion-3"
                    >
                      default
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
